### DISCUSSION TITLE
RS41 Connector

### DISCUSSION THREAD

**Discussion by Marjo** [mikaelnousiainen/RS41ng#30] [Q&A]
Hi! I got a sonde and were hoping to flash it with your firmware, problem is that i cant find a good connector (XData socket) for it, do you use home built ones or did you buy it some were?

**Comment by KB8RCO** [comment]
I bought Uxcell 10-conductor, 2mm ribbon cable from Amazon (pkg of 5 for < $10).
Cut 1 in half 
The 1 side for serial interfacing 
   Connect ribbon wires: 1, 2, 3, 5 (Gnd, Rx, Tx, Vcc )  to a DE9F pins: 5, 2, 3, 4 (Gnd, Tx, Rx, Vcc).  
   Although TTL levels (not RS-232) the DE9F pinout should match other devices attached to a serial port.
   This matches other APRS devices I have - all of which are set to TTL levels (typically remove Max232 chip and jumper for TTL)
The other side (not the same) I splice in a normal (0.1-inch) ribbon cable to match the ST-Link/V2.
   Connect ribbon wires:  6, 8, 9, 10 (Vcc, SWCLK, SWDIO, Gnd)  to a other ribbon cable wires: 7, 6, 2, 3 (Gnd, Tx, Rx, Vcc).  
I have not used the ST-Link/V2 yet, but the serial connection works well
---
I hope this help.
Rob KB8RCO

**Comment by mikaelnousiainen** [comment]
@Marjo There are indeed connectors for the "2mm" pin header available in eBay, for example this one:

https://www.ebay.com/itm/112062901292

Generally look for 5x2 2mm female connectors.

It is still possible to use individual pieces of the more common 0.1-inch/100mil/2.54mm crimp connectors here (especially if you remove the plastic "housing"), as only 4 pins need to be connected for programming.
